subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -649,4 +647,19 @@ public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setPropertyUpgradeReportEnabled(true);
         }
     }
+
+    public static class ProblemReportGenerationOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
+
+        public static final String LONG_OPTION = "disable-problem-report";
+        public static final String GRADLE_PROPERTY = "org.gradle.problem.report.disabled";
```

Comment:
❌ I think we can improve the property a little bit to fit better with the existing gradle properties.  I propose following:
- let's not use negative terms
- the property key should be org.gradle.problem.report with possible values `enabled` and `disabled` (or `true` and `false)
- the default value should be set to true
- we should have two command-line options: `--problem-report` and `--no-problem-report`.